native/libcst/src/tokenizer/whitespace_parser.rs

```diff
@@ -231,29 +224,34 @@ pub fn parse_empty_lines<'a>(
 
 pub fn parse_comment<'a>(config: &Config<'a>, state: &mut State) -> Result<Option<Comment<'a>>> {
     let newline_after = config.get_line_after_column(state.line, state.column_byte)?;
-    if let Some(comment_match) = COMMENT_RE.with(|r| r.find(newline_after)) {
-        let comment_str = comment_match.as_str();
-        advance_this_line(
-            config,
-            state,
-            comment_str.chars().count(),
-            comment_str.len(),
-        )?;
-        return Ok(Some(Comment(comment_str)));
+    if newline_after.is_empty() || newline_after.as_bytes()[0] != b'#' {
+        return Ok(None);
     }
-    Ok(None)
+    let comment_str = if let Some(idx) = memchr2(b'\n', b'\r', newline_after.as_bytes()) {
+        &newline_after[..idx]
+    } else {
+        newline_after
+    };
+    advance_this_line(
+        config,
+        state,
+        comment_str.chars().count(),
+        comment_str.len(),
+    )?;
+    Ok(Some(Comment(comment_str)))
 }
 
 pub fn parse_newline<'a>(config: &Config<'a>, state: &mut State) -> Result<Option<Newline<'a>>> {
     let newline_after = config.get_line_after_column(state.line, state.column_byte)?;
-    if let Some(newline_match) = NEWLINE_RE.with(|r| r.find(newline_after)) {
-        let newline_str = newline_match.as_str();
-        advance_this_line(
-            config,
-            state,
-            newline_str.chars().count(),
-            newline_str.len(),
-        )?;
+    let len = match newline_after.as_bytes() {
```

Comment:
I was interested in the performance characteristics of this vs other ways of checking a string prefix. With [this benchmark script](https://gist.github.com/orf/d4473dd0dff2bd480680b869f56929d6), 

```rust
match &thing.get(..2) {
        Some("\n") => 1,
        Some("\r\n") => 2,
        Some("\r") => 1,
        _ => 0,
    }
```

is a bit faster in all cases: `914` picoseconds vs `680` picoseconds for the `\r\n` case. Not that it makes much difference, but it is interesting because I would assume they optimize to the same code.

Reply:
Ahh, that's not exactly equivalent, it would return `None` on single digit characters. Ignore me.
```